### PR TITLE
 #162948833 Heroku deployment

### PR DESCRIPTION
**Pull request by d-kahara**
**What does this PR do?**
Setup heroku app configurations for staging and production environments

**Description of Tasks to be completed?**

- Install development dependencies i.e django-heroku(automatically configures DATABASE_URL, 
  ALLOWED_HOSTS, WhiteNoise (for static assets), Logging, and Heroku CI for the application)
- Create a Procfile with necessary settings
- Setup pipelines for staging and production environments
- Configure postgres addon for heroku
- Integrate heroku with github repo 

#### How should this be manually tested?

- Navigate to this url: https://ah-django-unchained-staging.herokuapp.com/

**What are the relevant pivotal tracker stories**

[##162948833](https://www.pivotaltracker.com/story/show/162948833)

**Screenshots**
image

### CODE REVIEW

**Comment by kwameasiago** [comment]
Use git commit --no-edit instead of having two commits that have the same message.

**Review by andrewhingah — APPROVED** [review]
Well written commit message. Just consider using `git commit --no edit` to avoid having two commits with similar message.

**Review by kwameasiago — APPROVED** [review]
Looks good to me

**Review by MaggieChege — APPROVED** [review]
Look good to me as well.

File: authors/apps/authentication/serializers.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ class LoginSerializer(serializers.Serializer):
     email = serializers.CharField(max_length=255)
     username = serializers.CharField(max_length=255, read_only=True)
     password = serializers.CharField(max_length=128, write_only=True)
+
```
Comment (ignatiusukwuoma) [inline authors/apps/authentication/serializers.py]: Why a blank line addition here?

File: authors/settings.py [hunk 2]
```diff
@@ -146,3 +148,5 @@
     #     'authors.apps.authentication.backends.JWTAuthentication',
     # ),
 }
+django_heroku.settings(locals())
+
```
Comment (ignatiusukwuoma) [inline authors/settings.py:150]: Why add a blank line after every piece of code?

**Comment by ignatiusukwuoma** [comment]
Rebase this branch on master carefully and ensure that no package in the requirements.txt is deleted

File: authors/settings.py (hunk 2 above)
Comment (d-kahara) [reply authors/settings.py:150]: Error on my part, I had misread the PEP 8 guidelines on [blank](https://www.python.org/dev/peps/pep-0008/#blank-lines)  lines.

File: requirements.txt [hunk 3]
```diff
@@ -5,4 +5,10 @@ djangorestframework==3.9.0
...
 psycopg2-binary==2.7.6.1
-pycodestyle==2.4.0
\ No newline at end of file
+pycodestyle==2.4.0
+django-heroku==0.3.1
+dj-database-url==0.5.0
+pytz==2018.9
+whitenoise==4.1.2
+gunicorn==19.9.0
+django-heroku==0.3.1
```
Comment (ignatiusukwuoma) [inline requirements.txt:13]: We have django-heroku twice in this file

File: authors/apps/authentication/serializers.py (hunk 1 above)
Comment (ignatiusukwuoma) [reply authors/apps/authentication/serializers.py]: Remove unnecessary white-spaces

**Comment by ignatiusukwuoma** [comment]
Are we supposed to have DEBUG=True in production?

File: requirements.txt (hunk 3 above)
Comment (d-kahara) [reply requirements.txt:13]: > Are we supposed to have DEBUG=True in production?

We'll change this when we're ready to ship for production?

File: authors/apps/authentication/serializers.py (hunk 1 above)
Comment (ignatiusukwuoma) [reply authors/apps/authentication/serializers.py]: Please remove your edit on this file. Your task doesn't require you to edit this file

**Comment by ignatiusukwuoma** [comment]
Your Travis build should pass now since all tests are passing

**Comment by ignatiusukwuoma** [comment]
Let me know when this PR is ready to be merged

**Comment by d-kahara** [comment]
> Let me know when this PR is ready to be merged

Ready to merge

**Pull request merged**